### PR TITLE
WT-6066 Enable datafile test

### PR DESCRIPTION
**Pull request by hbokhari**
The aim is to enable the tests for Evergreen testing. There are two major changes:
- `format` doesn't support column store configs at the moment and therefore I have disabled them by explicitly adding row-store config.
- Although we have logging selected in config, `format` can override that config if it's incompatible with other randomly selected config options. I have modified the script to look for log files only if they were created.

### CODE REVIEW

**Comment by lukech** [comment]
@hbokhari good finding on the cause of the issue. It feels not so right that the explicitly set `logging=1` being overwritten to `logging=0` due to incompatibility with certain random generated config options. My intuition is those explicitly-set config options should be honored over random ones. @keithbostic can you share your opinion on this?

Also, since we have an ongoing project aiming to re-enable column store support, should we create a follow-on ticket to get the column store re-enabled for endiness tasks, and make sure it's being part of the project scope?

**Comment by keithbostic** [comment]
> It feels not so right that the explicitly set logging=1 being overwritten to logging=0 due to incompatibility with certain random generated config options. My intuition is those explicitly-set config options should be honored over random ones.

I agree that explicitly-set config options should be honored in all cases. Generally, format is written that goal in mind, that is, explicitly set config options either cause failure or are honored (that's the reason for the `isperm` flag that's used throughout the `config.c` format sources).

Further, I think this specific change weakens the test, that is, because we could pass the test without creating log files, we might not detect problems in log file endianness.

**Comment by keithbostic** [comment]
@hbokhari, it looks to me like `prepare` configuration is the problem, and this change seems to fix the problem for me:
```
diff --git a/test/format/config.c b/test/format/config.c
index 996d63634..3911ec3ed 100644
--- a/test/format/config.c
+++ b/test/format/config.c
@@ -953,6 +953,11 @@ config_transaction(void)
         if (g.c_txn_freq != 100 && config_is_perm("transaction.frequency"))
             testutil_die(EINVAL, "prepare requires transaction frequency set to 100");
     }
+    if (g.c_logging && config_is_perm("logging")) {
+       if (g.c_prepare)
+                config_single("ops.prepare=off", false);
+    }
+
     if (g.c_txn_timestamps && config_is_perm("transaction.timestamps")) {
         if (g.c_isolation_flag != ISOLATION_SNAPSHOT && config_is_perm("transaction.isolation"))
             testutil_die(EINVAL, "timestamps require snapshot isolation");
```

**Comment by hbokhari** [comment]
@keithbostic, thank you for providing insight. I believe in addition to prepare, we also need to look into why rollback-to-stable disables logging as well:

**Comment by kommiharibabu** [comment]
> @keithbostic, thank you for providing insight. I believe in addition to prepare, we also need to look into why rollback-to-stable disables logging as well -
> 
> https://github.com/wiredtiger/wiredtiger/blob/9cd565809ead7473fea0089d943cf323e1368281/test/format/config.c#L994

RTS doesn't operate on logged tables as these updates are stable when the transaction is committed. I feel due to this reason may be the logging is disabled.

**Comment by keithbostic** [comment]
@hbokhari, is there no reasonable configuration in WiredTiger where both RTS & logging will be configured?

**Comment by hbokhari** [comment]
Thanks @kommiharibabu why we don't configure RTS and logging at the same time.

@keithbostic, I can't any permanent config that have both RTS and logging set. I'll try a change that will skip setting RTS when logging in enabled permanently.

**Comment by keithbostic** [comment]
@hbokhari, I'm not sure what you mean by `I can't any permanent config that have both RTS and logging set.`

Can you confirm there is no reasonable configuration of RTS and logging? In which case, shouldn't we disallow the combination in the library itself?

**Comment by hbokhari** [comment]
@keithbostic I meant that I can't find any existing format configs for the tests have both RTS and logging set explicitly in the config.

Looking at the comment from @kommiharibabu, there is no need to test RTS with logging as RTS skips logged tables. So yes it makes sense to disallow the combination in the library itself.

**Comment by keithbostic** [comment]
@hbokhari, what I'm trying to understand is if there is any application that would reasonably configure both logging and RTS. Our not testing that configuration in existing format configs, or that RTS skips logged tables doesn't resolve that question for me. If it's a reasonable configuration combination, we should be testing it, if it's not reasonable, the library should disallow it.

@kommiharibabu, can we get a definitive answer from you?

**Comment by kommiharibabu** [comment]
@keithbostic Enabling both `logging` and `RTS` is a valid configuration and MongoDB does the same configuration. But from the `test/format` point of view, in the above configuration, `RTS` don't operate on the `logged` table. If we are not expecting `RTS` to do anything from the test point of view, there is no problem in enabling the above configuration. If the test is expecting something that `RTS` do, then the test needs to be fixed before enabling.

**Comment by keithbostic** [comment]
Thank you, @kommiharibabu. Based on your answer, I would argue for removing the constraint from `format` and seeing if anything breaks.

File: test/format/CONFIG.endian [hunk 1]
```diff
@@ -4,3 +4,4 @@ logging.archive=0
 logging=1
 runs.timer=4
 runs.rows=1000000
+runs.type=row-store
```
Comment (keithbostic) [inline test/format/CONFIG.endian:7]: This is another change that should be reverted if we ever make column-store work again.

Comment (hbokhari) [reply test/format/CONFIG.endian:7]: I'll create a ticket to revert the change once we have column-store working with `format`.

**Comment by hbokhari** [comment]
I have created WT-7294 to re-enable column-store testing.

File: test/format/config.c [hunk 2]
```diff
@@ -962,6 +962,12 @@ config_transaction(void)
     if (g.c_logging && config_is_perm("logging")) {
         if (g.c_prepare)
             config_single("ops.prepare=off", false);
+        /* Rollback-to-stable doesn't operate on logged tables. */
+        if (g.c_txn_rollback_to_stable && config_is_perm("transaction.rollback_to_stable"))
+            testutil_die(
+              EINVAL, "rollback to stable and logging can not be configured at the same time");
+        else if (g.c_txn_rollback_to_stable)
+            config_single("transaction.rollback_to_stable=off", false);
```
Comment (keithbostic) [inline test/format/config.c:970]: I don't understand the reason for this change -- why does it improve anything?

**Comment by hbokhari** [comment]
@keithbostic, sorry I missed pushing my final commit before asking for your review. Two main changes:

- Disable prepared updates if we have logging configuration is permanent. 
- Avoid disabling logging config if rollback to stable config is set. Based on our discussion we haven't found a strong reason to disable logging - let's see if there is any fall out though.

File: test/format/config.c (hunk 2 above)
Comment (hbokhari) [reply test/format/config.c]: I have replied [here](https://github.com/wiredtiger/wiredtiger/pull/6379#issuecomment-799034913)

**Review by keithbostic — APPROVED** [review]
lgtm

**Comment by hbokhari** [comment]
@lukech, can you have a look?

**Review by lukech — APPROVED** [review]
LGTM. Thanks for creating the follow-on ticket. Do you mind updating the PR & Jira title to something like "Re-enable endiness tests"? "datafile test" is a bit ambiguous to me.

**Comment by hbokhari** [comment]
@lukech - I have renamed the ticket name to "Re-enable endianness tests on evergreen"

**Pull request merged**